src/bluesearch/database/article.py

```diff
@@ -73,6 +87,163 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
+                # In rare cases, an author may not have a given name or a surname,
+                # e.g. it could be an organization. We decide to skip those.
+                continue
+            author_str = given_names + " " + surname
+            yield author_str.strip()
+
+    @property
+    def abstract(self) -> Generator[str, None, None]:
+        """Get a sequence of paragraphs in the article abstract.
+
+        Yields
+        ------
+        str
+            The paragraphs of the article abstract.
+        """
+        abstract_pars = self.content.findall(".//abstract//p")
+        for paragraph in abstract_pars:
+            yield self.text_content(paragraph)
+
+    @property
+    def paragraphs(self) -> Generator[Tuple[str, str], None, None]:
+        """Get all paragraphs and titles of sections they are part of.
+
+        Paragraphs can be parts of text body, or figure or table captions.
+
+        Yields
+        ------
+        section : str
+            The section title.
+        text : str
+            The paragraph content.
+        """
+        # Paragraphs of text body
+        abstract_p = self.content.findall(".//abstract/p")
+        caption_p = self.content.findall(".//caption/p")
+        acknowlegd_p = self.content.findall(".//ack/p")
+
+        exclude_list = abstract_p + caption_p + acknowlegd_p
+
+        section_dirs = self.get_sections()
+        for paragraph in self.content.findall(".//p"):
+            if paragraph not in exclude_list:
+                text = self.text_content(paragraph)
+                section_title = ""
+                if paragraph in section_dirs:
+                    section_title = section_dirs[paragraph]
+                yield section_title, text
+
+        # Figure captions
+        figs = self.content.findall(".//fig")
+        for fig in figs:
+            fig_captions = fig.findall("caption")
+            if fig_captions is None:
+                continue
+            caption = " ".join(self.text_content(c) for c in list(fig_captions))
+            yield "Figure Caption", caption
+
+        # Table captions
+        tables = self.content.findall(".//table-wrap")
+        for table in tables:
+            caption_elements = table.findall(".//caption/p") or table.findall(
+                ".//caption/title"
+            )
+            if caption_elements is None:
+                continue
+            caption = " ".join(self.text_content(c) for c in caption_elements)
+            yield "Table Caption", caption
+
+    def get_sections(self) -> dict:
+        """Extract sections information.
+
+        Returns
+        -------
+        sections_dir : dict
+            Dictionary whose keys are paragraphs and value are section title.
+        """
+        sections_dir = {}
+        for sec in self.content.findall(".//sec"):
+
+            section_title = ""
+            section_paragraphs = []
+
+            for element in list(sec):
```

Comment:
It seems `sec` is already an iterable, so maybe casting to `list` is not necessary?

Reply:
Done in d0db1e4